include/zephyr/net/socket.h

```diff
@@ -1095,10 +1095,33 @@ struct ifreq {
 /** sockopt: Set or receive the Type-Of-Service value for an outgoing packet. */
 #define IP_TOS 1
 
+/** sockopt: Pass an IP_PKTINFO ancillary message that contains a
+ *  pktinfo structure that supplies some information about the
+ *  incoming packet.
+ */
+#define IP_PKTINFO 8
+
+struct in_pktinfo {
+	unsigned int   ipi_ifindex;  /* Interface index */
+	struct in_addr ipi_spec_dst; /* Local address */
+	struct in_addr ipi_addr;     /* Header Destination address */
+};
+
 /* Socket options for IPPROTO_IPV6 level */
 /** sockopt: Don't support IPv4 access (ignored, for compatibility) */
 #define IPV6_V6ONLY 26
 
+/** sockopt: Pass an IPV6_RECVPKTINFO ancillary message that contains a
+ *  in6_pktinfo structure that supplies some information about the
+ *  incoming packet. See RFC 3542.
+ */
+#define IPV6_RECVPKTINFO 49
+
+struct in6_pktinfo {
+	struct in6_addr ipi6_addr;    /* src/dst IPv6 address */
```

Comment:
When is it src and when is it dst address? It seems we only set this as a dst address on receive?

Reply:
I copied the struct verbatim from the RFC. I think the struct can be used in other ways too (did not check this).